doc/developer_guide/dependency.rst

```diff
@@ -0,0 +1,458 @@
+else-body. It is the responsibility of the user to handle these cases - for
+example by creating separate `VariablesAccessInfo` for statements in the if-body
+and for the else-body.
+
+.. note:: When using different instances for an if- and else-body, the first
+    statement of the if-body will
+    have the same location number as the first statement of the else-body. So
+    you can only compare location numbers from the same `VariablesAccessInformation`
+    instance. If you merge two instances together, the locations of the merged-in
+    instance will be appropriately increased to follow the locations of the
+    instance to which it is merged.
+
+
+The location number is not exactly a line number - several statements can be
+on one line, which will get different location numbers. And certain lines
+will not have a location number (e.g. comment lines).
+
+As stated above, one instance of `VariablesAccessInfo` can be extended by adding
+additional variable information. It is the responsibility of the user to make
+sure the accesses are added in the right order - the `VariablesAccessInfo` object
+will always assume accesses happen at the current location, and a call to 
+`next_location()` is required to increase the location number.
+
+.. note:: It is not possible to add access information about an earlier
+     usage to an existing `VariablesAccessInfo` object. 
+
+
+Access Examples
+---------------
+
+Below we show a simple example of how to use this API. This is from the
+`psyclone.psyGen.OMPParallelDirective` (so `self` is an instance of this
+node), and this code is used to determine a list of all the scalar
+variables that must be declared as thread-private::
+
+  var_accesses = VariablesAccessInfo()
+  self.reference_accesses(var_accesses)
+  for signature in var_accesses.all_signatures:
+      var_name = str(signature)
+      access_info = var_accesses[signature]
+      symbol = symbol_table.lookup(var_name)
+
+      # Ignore variables that are arrays, we only look at scalar ones.
+      # The `is_used_as_array` function will take information from
+      # the access information as well as from the symbol table
+      # into account.
+      is_array = symbol.is_used_as_array(access_info=acess_info)
+
+      if is_array:
+          # It's not a scalar variable, so it will not be private
+          continue
+
+      # If a scalar variable is only accessed once, it is either a coding
+      # error or a shared variable - anyway it is not private
+      accesses = var_accesses[signature].all_accesses
+      if len(accesses) == 1:
+          continue
+
+      # We have at least two accesses. If the first one is a write,
+      # assume the variable should be private:
+      if accesses[0].access_type == AccessType.WRITE:
+          result.add(var_name.lower())
+
+The next, hypothetical example shows how the `VariablesAccessInfo` class
+can be used iteratively. Assume that you have a function that determines
```

Comment:
I think we do actually have such a function now?

Reply:
Well, yes, and we always had. This is just a coding example that can be used as a kind of template on how to use the access information.